apps/ledger-live-desktop/src/renderer/hooks/nfts/useNftCollectionsStatus.ts

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from "react";
+import { useSelector } from "react-redux";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { nftCollectionsStatusByNetworkSelector } from "~/renderer/reducers/settings";
+import { NftStatus } from "@ledgerhq/live-nft/types";
+
+export function useNftCollectionsStatus() {
+  const nftsFromSimplehashFeature = useFeature("nftsFromSimplehash");
+  const nftCollectionsStatusByNetwork = useSelector(nftCollectionsStatusByNetworkSelector);
+
+  const shouldDisplaySpams = !nftsFromSimplehashFeature?.enabled;
+
```

Comment:
Both `list` and `whitelisted` have the same logic process except for the `.filter(...)`  business logic. In this case and to avoid code duplication i suggest a generic function with dependency inversion in the status filtering part. 
The generic function can be easily extracted from the hook as a helper, but it's ok if you want to keep it her
```typescript 
// this is the generic function (that parses the nftCollection) 
const neftCollectionParser = (nftCollection, appltFilterFn) => Object.values(nftCollection).flatMap(contracts =>
      Object.entries(contracts)
        .filter(appltFilterFn)
        .map(([contract]) => contract),
    );

const list = useMemo(() => {
  neftCollectionParser(nftCollectionsStatusByNetwork,([_, status]) =>
          !shouldDisplaySpams ? status !== NftStatus.whitelisted : status === NftStatus.blacklisted)
},[nftCollectionsStatusByNetwork, shouldDisplaySpams]);

const whitelisted = useMemo(() => {
  neftCollectionParser(nftCollectionsStatusByNetwork,([_, status]) => status === NftStatus.whitelisted)
},[nftCollectionsStatusByNetwork, shouldDisplaySpams]);
```

Reply:
the benefit from extracting `neftCollectionParser` in a helper function , is that you can use it in other part of the app (for instance : in `orderedVisibleNftsSelector`)